### PR TITLE
[Xamarin.Android.Build.Tasks] Handle Empty R.txt file.

### PR DESCRIPTION
**Pull request by dellis1972**
Fixes https://i.azdo.io/1439405

The ManagedResourceParser was not handling the case where we do not
get four items per line. As a result it would crash with the following
error.

```
System.IndexOutOfRangeException: Index was outside of bounds of the array.
as Xamarin.And.Tasks.ManagedResourceParser.ProcessRtxtFile (System.String file)
```

If we do not get the expected four items from the line in the file we
should ignore that line. A unit test has been added to check that the
task does not fail when this happens.

### CODE REVIEW

File: src/Xamarin.Android.Build.Tasks/Utilities/ManagedResourceParser.cs [hunk 1]
```diff
@@ -299,6 +299,8 @@ void ProcessRtxtFile (string file)
 			var lines = System.IO.File.ReadLines (file);
 			foreach (var line in lines) {
 				var items = line.Split (new char [] { ' ' }, 4);
+				if (items.Length < 4)
+					continue;
```
Comment (jonathanpeppers) [inline src/Xamarin.Android.Build.Tasks/Utilities/ManagedResourceParser.cs:303]: Is it worth logging here? `LogDebugMessage()` with the file path?

File: src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/Tasks/ManagedResourceParserTests.cs [hunk 2]
```diff
@@ -426,6 +426,19 @@ public void GenerateDesignerFileFromRtxt ([Values (false, true)] bool withLibrar
...
 		}
 
+		[Test]
+		public void GenerateDesignerFileFromEmptyRtxt ()
+		{
+			var path = Path.Combine ("temp", TestName + " Some Space");
+			CreateResourceDirectory (path);
+			var task = CreateTask (path);
+			task.RTxtFile = Path.Combine (Root, path, "R.txt");
+			File.WriteAllText (task.RTxtFile, string.Empty);
```
Comment (jonpryor) [inline src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/Tasks/ManagedResourceParserTests.cs:436]: This test isn't testing what you think it is.  This results in a 0-byte file, and with a 0-byte input, `File.ReadLines()` returns a sequence with 0 elements, meaning we won't actually hit the error condition.

What we *actually* want is a file containing a *blank line*:

```csharp
File.WriteAllText (task.RTxtFile, Environment.NewLine);
```

*Then* `File.ReadLines()` will return a non-empty sequence, and the first (and only) value it returns will be `string.Empty`, and `string.Empty.Split(new char[]{' '}, 4)` returns `new string[]{""}` (a 1 element array), which triggers the error scenario.

File: src/Xamarin.Android.Build.Tasks/Utilities/ManagedResourceParser.cs [hunk 3]
```diff
@@ -299,6 +299,10 @@ void ProcessRtxtFile (string file)
 			var lines = System.IO.File.ReadLines (file);
 			foreach (var line in lines) {
 				var items = line.Split (new char [] { ' ' }, 4);
+				if (items.Length < 4) {
+					Log.LogDebugMessage ($"Ignoring '{line}' from '{file}'. It does not have the correct number of elements.");
```
Comment (jonpryor) [inline src/Xamarin.Android.Build.Tasks/Utilities/ManagedResourceParser.cs:303]: Would it be useful to know the line number?  That would be extra work to track, but it *might* be useful to know if this is the 1st line or the 1000th line…?

Comment (jonpryor) [reply src/Xamarin.Android.Build.Tasks/Utilities/ManagedResourceParser.cs:303]: Could result in a debug message of:

```
{file}:{line_number}: ignoring contents `{line}`, as it does not have the correct number of elements.
```

**Pull request merged**